docs/pages/server-access/rbac.mdx

```diff
@@ -0,0 +1,182 @@
+granular access policies for Linux servers connected to Teleport.
+
+An example of a policy could be, *""server administrators have access to
+everything, QA team and engineers have full access to staging servers, and
+engineers can gain temporary access to the production server in case of
+emergency"*.
+
+For a more general description of Teleport roles and examples see [RBAC](../access-controls/introduction.mdx), as
+this section focuses on configuring RBAC for Server Access.
+
+## Role configuration
+
+Teleport's "role" resource provides the following instruments for restricting
+server access:
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: developer
+spec:
+  allow:
+    # The logins array defines the OS/UNIX logins a user is allowed to use.
+    # both strings and template variables are supported in this field
+    logins: [ubuntu, debian, '{{internal.logins}}']
+
+    # node_labels: a user with this role will be allowed to connect to
+    # SSH nodes whose labels match any of the expressions below.
+    node_labels:
+      # literal strings:
+      'env': 'test'
+      # the wildcard ('*') means "any node"
+      '*': '*'
+      # a list of allowed values:
+      'region': ['us-west-1', 'eu-central-1']
+      # regular expressions start with ^ and end with $
+      # Teleport uses golang regexp syntax.
+      # the list example above can be expressed as:
+      'reg': '^us-west-1|eu-central-1$'
+
+    # List of host groups the created user will be added to. Any that don't already exist are created.
+    host_groups: [ubuntu, nginx, other]
+
+    # Assign the user to the sudoers group
+    host_sudoers:
+    - 'ALL=(ALL) NOPASSWD: ALL'
+
+    # Deny access to the root user
+  deny:
+    logins:
+    - root
+```
+
+<Admonition
+  type="note"
+  title="Deny Rules"
+>
+  Deny rules will match greedily. In the example above, a server session
+  attempting to use "root" serveruser  account will be rejected.
+</Admonition>
+
+## Template variables
+
+Similar to other role fields, the Server related fields support templating variables.
+
+The `{{external.xyz}}` variables are replaced with values from external [SSO](../access-controls/sso.mdx)
+providers. For OIDC logins, `{{external.xyz}}` refers to the "xyz" claim; for
+SAML logins, `{{external.xyz}}` refers to the "xyz" assertion.
+
+For example, here is what a role may look like if you want to assign allowed
+server environment types and allowed logins from the user's Okta `environments` and 
+`allowedlogins` assertions:
+
+```yaml
+spec:
+  allow:
+    node_labels:
+    - env:  '{{external.environments}}'
+    logins:
+    - '{{external.allowedlogins}}'
+```
+
+The `{{internal.logins}}` variable permits sharing
+allowed server logins with remote clusters. They will be replaced
+with the respective properties of a remote user connecting from a root cluster.
+
+For example, suppose a user in the root cluster has the following role:
```

Comment:
We haven't introduced the term "root cluster" yet on this page. I think it would make sense to mention Trusted Clusters here if we're going to use the term "root cluster". The `internal` prefix also works without Trusted Clusters, though, right?